prost-build/src/code_generator.rs

```diff
@@ -883,8 +876,10 @@ impl CodeGenerator<'_> {
                 let name = method.name.take().unwrap();
                 let input_proto_type = method.input_type.take().unwrap();
                 let output_proto_type = method.output_type.take().unwrap();
-                let input_type = self.resolve_ident(&input_proto_type);
-                let output_type = self.resolve_ident(&output_proto_type);
+                let input_type =
+                    self.resolve_ident(&FullyQualifiedName::from_type_name(&input_proto_type));
```

Comment:
Does it make sense to `impl From<&str> for FullyQualifiedName`? Then this could do: `input_proto_type.into()`.